aptos-move/framework/aptos-framework/sources/object.spec.move

```diff
@@ -1,9 +1,5 @@
 spec aptos_framework::object {
-    spec exists_at {
-        pragma opaque;
-        aborts_if false;
-        // TODO: Disabled the following post-condition due to an issue with
-        // the use of a type parameter in `exists` in spec.
-        // ensures result == exists<T>(object);
```

Comment:
@wrwg, the line `ensures result == exists<T>(object);` currently has an issue. I am not sure if it's a bug or a design choice.

Currently, in the exists expression `exists<T>(a)`, `T` should be a concrete struct type like `CoinStore`. Even if `T` is a type parameter with the key-ability, Prover will complain: https://github.com/aptos-labs/aptos-core/issues/6658.

It seems that this behavior is consistent in the code (not only spec) as well. So, the following code isn't allowed currently:
```
module 0x42::exists_at {
    fun exists_at<T: key>(a: address): bool {
        exists<T>(a)
    }
}

error: expected specific type
  ┌─ ./sources/exists_at.move:4:9
  │
4 │         exists<T>(a)
  │         ^^^^^^^^^^^^
  │         │      │
  │         │      Expected a struct type. Global storage operations are restricted to struct types declared in the current module. Found the type parameter: 'T'
  │         Invalid call to exists.
```
(cc: @meng-xu-cs )

Reply:
We attempted to natively define a spec function like this:
```
spec native fun spec_exists_at<T>(object: address): bool;
```
It was not straightforward because a spec function cannot be state-dependent (referring to the memory variables).

Reply:
I see. This is actually a restriction of the Move compiler: https://github.com/move-language/move/blob/e1ad2a278ace585ddc492b8bde2c861e9ce0861a/language/move-compiler/src/typing/globals.rs#L219, not the prover.

Maybe the v2 compiler can deal with this in a better way.